### PR TITLE
Vendor rack-protection/authenticity_token middleware

### LINKED ISSUE #4587: Remove dependency on `rack-protection`?
Can we remove the dependency on `rack-protection`? afaik we only use the CSRF protection. Can we vendor that?

### PR DESCRIPTION
**Pull request by seuros**
Closes #4587

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -211,3 +208,6 @@ DEPENDENCIES
   sqlite3
   standard
   toxiproxy
+
```
Comment (mperham) [inline Gemfile.lock:211]: Remove this and Bundler version doesn’t matter

Comment (seuros) [reply Gemfile.lock]: Done.

File: lib/sidekiq/web/authenticity_token.rb [hunk 2]
```diff
@@ -0,0 +1,231 @@
+require 'securerandom'
+require 'base64'
+require 'rack/request'
+
+module Sidekiq
+  class Web
+    ## Shamelessly copied from Rack-protection gem
```
[the remark this answers is not in the document]
Comment (mperham) [reply lib/sidekiq/web/authenticity_token.rb:7]: https://github.com/sinatra/sinatra/blob/master/rack-protection/License

**Comment by mperham** [comment]
Well done! I wonder if we could remove Rack and just have Sidekiq::Web quack like a Rack app... 🤔

**Comment by natematykiewicz** [comment]
> Well done! I wonder if we could remove Rack and just have Sidekiq::Web quack like a Rack app... 🤔

A safer option is probably to add a soft dependency on Rack. Like how Rails does with various database drivers.

https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L3-L4

Basically, you still require a specific Rack version _**if**_ they attempt to use mount the Web UI.

**Pull request merged**